VideoLocker/src/main/java/org/edx/mobile/core/ServerJsonDateAdapterFactory.java

```diff
@@ -58,24 +58,29 @@ public void write(JsonWriter out, Date value) throws IOException {
 
         @Override
         public Date read(JsonReader in) throws IOException {
-            String dateString = in.nextString();
+            if (in.peek() == JsonToken.NULL) {
+                in.nextNull();
+                return null;
```

Comment:
Note that there is a `nullSafe()` [method][nullSafe() method] in `TypeAdapter` that wraps it to not having any null tokens sent to the implementation, to avoid this boilerplate handling in each implementation.